scripts/testing/database-emulator-test.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const { spawn } = require('child-process-promise');
+
+const EMULATOR_LINK =
+  'https://firebase.google.com/docs/database/downloads/database-emulator.jar';
```

Comment:
cc/ @ryanpbrewster 

I'm surprised this link isn't versioned in any way (i.e. I'd expect database-emulator-0.1.jar or something).  As-is, if the emulator ever makes a breaking change, we'll be broken.  Has this been discussed at all?

Reply:
It's using a link that specifies the version now.

It's been discussed in this [doc](https://docs.google.com/document/d/1k_nmMPL9LRFzxk9yJceNYgHcxgiMeIaAc07YfKUgtqk/edit) a bit. The emulator version is tied to firebase cli version, if we do download and startup through cli. After we have addresses thoughts in the doc, we will move to firebase cli eventually.